executor/prepared.go

```diff
@@ -121,7 +122,15 @@ func (e *PrepareExec) DoPrepare() {
 		}
 	}
 	charset, collation := variable.GetCharsetInfo(e.Ctx)
-	stmts, err := parser.Parse(e.SQLText, charset, collation)
+	var (
+		stmts []ast.StmtNode
+		err   error
+	)
+	if sqlParser, ok := e.Ctx.(sqlexec.SQLParser); ok {
+		stmts, err = sqlParser.ParseSQL(e.SQLText, charset, collation)
+	} else {
```

Comment:
if one day context change to other implement, or been wrapped in another struct, we won't meet a strange panic.
It's not good to make strong assumption of a particular implementation.